huaweicloud/services/evs/resource_huaweicloud_evs_snapshot_group.go

```diff
@@ -0,0 +1,517 @@
+			return false, fmt.Errorf("unmarshal the response body failed: %s", jsonErr)
+		}
+		errorCode, errorCodeErr := jmespath.Search("error_code||errCode", apiError)
+		if errorCodeErr != nil {
+			return false, fmt.Errorf("error parse errorCode from response body: %s", errorCodeErr)
+		}
+		if errorCode == "EVS.2409" {
+			return true, err
+		}
+	}
+	if errCode, ok := err.(golangsdk.ErrUnexpectedResponseCode); ok && errCode.Actual == 409 {
+		return true, err
+	}
+	return false, err
+}
+
+func resourceEvsSnapshotGroupRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v5/{project_id}/snapshot-groups/{snapshot_group_id}"
+		product = "evs"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating EVS client: %s", err)
+	}
+	requestPath := client.Endpoint + httpUrl
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	requestPath = strings.ReplaceAll(requestPath, "{snapshot_group_id}", d.Id())
+	requestOpt := golangsdk.RequestOpts{KeepResponseBody: true}
+	resp, err := client.Request("GET", requestPath, &requestOpt)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving EVS snapshot group")
+	}
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	group := utils.PathSearch("snapshot_group", respBody, nil)
+
+	mErr := multierror.Append(
+		d.Set("server_id", utils.PathSearch("server_id", group, nil)),
+		d.Set("name", utils.PathSearch("name", group, nil)),
+		d.Set("description", utils.PathSearch("description", group, nil)),
+		d.Set("enterprise_project_id", utils.PathSearch("enterprise_project_id", group, nil)),
+		d.Set("tags", utils.PathSearch("tags", group, map[string]interface{}{})),
+		d.Set("created_at", utils.PathSearch("created_at", group, nil)),
+		d.Set("status", utils.PathSearch("status", group, nil)),
+		d.Set("updated_at", utils.PathSearch("updated_at", group, nil)),
+	)
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceEvsSnapshotGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v5/{project_id}/snapshot-groups/{snapshot_group_id}"
+		product = "evs"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating EVS client: %s", err)
+	}
+	requestPath := client.Endpoint + httpUrl
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	requestPath = strings.ReplaceAll(requestPath, "{snapshot_group_id}", d.Id())
+	requestOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildUpdateSnapshotGroupBodyParams(d)),
+	}
+	_, err = client.Request("PUT", requestPath, &requestOpt)
```

Comment:
If the user only changes the tags field, will this part make meaningless interface calls? Please use d.HasChangs to handle it in advance.